internal/p2p/peermanager.go

```diff
@@ -527,6 +591,17 @@ func (m *PeerManager) DialFailed(ctx context.Context, address NodeAddress) error
 
 	addressInfo.LastDialFailure = time.Now().UTC()
 	addressInfo.DialFailures++
+
+	var totalDialFailures uint32
+	for _, addr := range peer.AddressInfo {
+		totalDialFailures += addr.DialFailures
+	}
+
+	if m.options.MaxFailedDialAttempts > 0 && totalDialFailures > m.options.MaxFailedDialAttempts {
+		peer.Inactive = true
+		m.metrics.PeersInactivated.Add(1)
+	}
```

Comment:
Can we add some comments here. There's also a FIXME above that I think we can remove